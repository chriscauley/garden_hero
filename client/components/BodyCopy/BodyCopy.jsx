import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { buildClassName } from 'utils/util';
import { bodyCopyProps } from 'utils/propTypeEnum';

class BodyCopy extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node.isRequired,
    element: PropTypes.string,
    modifier: PropTypes.oneOf(bodyCopyProps.modifiers),
    size: PropTypes.oneOf(bodyCopyProps.sizes)
  };

  static defaultProps = {
    element: 'span'
  };

  render() {
    const {
      children,
      className,
      element: Element,
      modifier,
      size,
      ...props
    } = this.props;
    const mainClass = 'bcp-body-copy';
    const modifierClass = buildClassName(
      bodyCopyProps.modifiers,
      modifier,
      mainClass
    );
    const sizeClass = buildClassName(bodyCopyProps.sizes, size, mainClass);

    return (
      <Element
        className={classNames(mainClass, modifierClass, sizeClass, className)}
        {...props}
      >
        {children}
      </Element>
    );
  }
}

export default BodyCopy;
