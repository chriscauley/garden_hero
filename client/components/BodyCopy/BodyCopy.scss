.gh-body-copy {
  @include body-copy;
  font-weight: $fontWeightRegular;
  margin-bottom: em(10px);
  word-break: break-word;
  &--s {
    @include heading-s;
  }
  &--l {
    @include heading-l;
  }
  &--block {
    display: block;
    margin-bottom: 0;
  }
  &--bold {
    font-weight: $fontWeightBold;
  }
  &--green {
    color: $green;
  }
}