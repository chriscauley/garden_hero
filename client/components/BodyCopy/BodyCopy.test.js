import React from 'react';
import renderer from 'react-test-renderer';

import BodyCopy from './BodyCopy';

describe('<BodyCopy />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<BodyCopy>Body copy text</BodyCopy>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts className prop', () => {
    const component = renderer.create(
      <BodyCopy className="gh-margin-top">Body copy text</BodyCopy>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts modifier prop', () => {
    const component = renderer.create(
      <BodyCopy modifier="bold">Body copy text</BodyCopy>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders small text', () => {
    const component = renderer.create(
      <BodyCopy size="s">Body copy text</BodyCopy>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
