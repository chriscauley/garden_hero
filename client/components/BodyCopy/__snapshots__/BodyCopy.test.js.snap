// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyCopy /> accepts className prop 1`] = `
<span
  className="bcp-body-copy bcp-margin-top"
>
  Body copy text
</span>
`;

exports[`<BodyCopy /> accepts modifier prop 1`] = `
<span
  className="bcp-body-copy bcp-body-copy--bold"
>
  Body copy text
</span>
`;

exports[`<BodyCopy /> renders correctly 1`] = `
<span
  className="bcp-body-copy"
>
  Body copy text
</span>
`;

exports[`<BodyCopy /> renders small text 1`] = `
<span
  className="bcp-body-copy bcp-body-copy--s"
>
  Body copy text
</span>
`;
